isis/tests/FunctionalTestsSpiceserver.cpp

```diff
@@ -0,0 +1,172 @@
+#include <QTextStream>
+#include <QStringList>
+#include <QTemporaryFile>
+#include <QByteArray>
+#include <QFile>
+#include <QVector>
+#include <QDomDocument>
+
+#include "spiceserver.h"
+#include "Fixtures.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "TestUtilities.h"
+#include "TextFile.h"
+
+#include "gmock/gmock.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/spiceserver.xml").expanded();
+
+class TestPayload : public TempTestingFiles {
```

Comment:
In hindsight, replacing this fixture with something built from the already existing defaultCube seems like the better option to avoid the large binary in code...

Reply:
I think this big hex binary changes every time we do a big environment or OS update, so dynamically generating the payload instead of hard-coding it would be great.